### DISCUSSION TITLE
grpc loadbalance

### DISCUSSION THREAD

**Discussion by xiaozaiyuji** [open-telemetry/opentelemetry-java-instrumentation#11045] [Q&A]
![image](https://github.com/open-telemetry/opentelemetry-java/assets/4230320/7268243f-177d-4690-8f3c-55afe5ab91fa)
I have a confusion, grpc loadbalance with dns has a limit. As follows
https://github.com/grpc/grpc/issues/12295#issuecomment-382794204
In grpc java,loadbalance can be config by this document.
https://github.com/grpc/grpc-java/issues/1771
I did not find the way to config java instrument grpc loadbalance.
So,how the java otel agent to config loadbalance?

**Comment by jkwatson** [comment]
I don't think this is something that is currently supported outside of building an agent extension to configure your channel exactly how you wanted. To build this into java, we would probably want to make sure we had a specification for how it should be done, so it can be done across multiple language implementations the same way.

**Reply by jkwatson** [reply]
I'm going to move this to the instrumentation repository, since it seems specific to doing configuration of the java agent, rather than the SDK itself via programmatic configurations.

**Comment by laurit** [comment]
The default grpc exporter from https://github.com/open-telemetry/opentelemetry-java does not use https://github.com/grpc/grpc-java If you wish to use `setChannel` you will have to build and [extension](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension) or a [custom distro](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/distro) that configures the grpc exporter to use grpc-java.
Alternatively, instead of using grpc, you could use http/protobuf that should be easier to load balance.
